src/OpenTelemetry.AutoInstrumentation.StartupHook/RulesEngine/RuleEngine.cs

```diff
@@ -32,7 +32,8 @@ internal class RuleEngine
     {
         new OpenTelemetrySdkMinimumVersionRule(),
         new DiagnosticSourceRule(),
-        new InstrumentationAssemblyRule()
+        new InstrumentationAssemblyRule(),
+        new NativeProfilerDiagnosticsRule()
```

Comment:
Order matters here; is your plan to evaluate this rule at the end?

Reply:
That's actually good question. I think it should be after all checks that determine if auto instrumentation can work at all. 
So maybe the last place is good, otherwise if it's too early, there is no point to add such overhead (if it's going to crash anyway).